cns/middlewares/mock/mockSWIFTv2.go

```diff
@@ -0,0 +1,92 @@
+package middlewares
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/cns/types"
+	"github.com/Azure/azure-container-networking/crd/multitenancy/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	k8types "k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	errMTPNCNotReady  = errors.New("mtpnc is not ready")
+	errFailedToGetPod = errors.New("failed to get pod")
+)
+
+type MockSWIFTv2Middleware struct {
+	mtPodState map[string]*v1.Pod
+	mtpncState map[string]*v1alpha1.MultitenantPodNetworkConfig
+}
+
+func NewMockSWIFTv2Middleware() *MockSWIFTv2Middleware {
+	testPod1 := v1.Pod{}
+	testPod1.Labels = make(map[string]string)
+	testPod1.Labels[configuration.LabelSwiftV2] = "true"
+
+	testMTPNC1 := v1alpha1.MultitenantPodNetworkConfig{}
+	testMTPNC1.Status.PrimaryIP = "192.168.0.1"
+	testMTPNC1.Status.MacAddress = "00:00:00:00:00:00"
+	testMTPNC1.Status.GatewayIP = "10.0.0.1"
+	testMTPNC1.Status.NCID = "testncid"
+
+	return &MockSWIFTv2Middleware{
+		mtPodState: map[string]*v1.Pod{"testpod1namespace/testpod1": &testPod1},
+		mtpncState: map[string]*v1alpha1.MultitenantPodNetworkConfig{"testpod1namespace/testpod1": &testMTPNC1},
+	}
+}
+
+// validateMultitenantIPConfigsRequest validates if pod is multitenant
+// nolint
+func (m *MockSWIFTv2Middleware) ValidateIPConfigsRequest(req *cns.IPConfigsRequest) (respCode types.ResponseCode, message string) {
+	// Retrieve the pod from the cluster
+	podInfo, err := cns.UnmarshalPodInfo(req.OrchestratorContext)
+	if err != nil {
+		errBuf := fmt.Sprintf("unmarshalling pod info from ipconfigs request %v failed with error %v", req, err)
+		return types.UnexpectedError, errBuf
+	}
+	podNamespacedName := k8types.NamespacedName{Namespace: podInfo.Namespace(), Name: podInfo.Name()}
+	pod, ok := m.mtPodState[podNamespacedName.String()]
+	if !ok {
+		errBuf := fmt.Sprintf("failed to get pod %v with error %v", podNamespacedName, err)
+		return types.UnexpectedError, errBuf
+	}
+	// check the pod labels for Swift V2, enrich the request with the multitenant flag. TBD on the label
```

Comment:
Seems we are mixing logic for the labels on the Node which is "multitenant-enabled" (secondary nic exists), versus labels for the pod *requiring* multitenancy. Pod labels for swift-v2 were expected to be: 
kubernetes.azure.com/pod-network ==> this marks the pod as requiring multitenancy
kubernetes.azure.com/pod-network-instance

Reply:
So the label `kubernetes.azure.com/podnetwork-multi-tenancy-enabled` should only be used for the node which I get, but should the pod label be `kubernetes.azure.com/pod-network` or `kubernetes.azure.com/pod-network-instance`? I'm kinda confused about your comment formatting. Do you mean `kubernetes.azure.com/pod-network` should be used to label swift-v2 pods and `kubernetes.azure.com/pod-network-instance` should be used to label pods that have its secondary nic(s) exist?